src/Workspaces/Core/Portable/CodeStyle/CodeStyleOption.cs

```diff
@@ -77,37 +107,39 @@ public static CodeStyleOption<T> FromXElement(XElement element)
             return new CodeStyleOption<T>(value, notificationOption);
         }
 
-        private static Func<string, object> GetParser(string type)
+        private static Func<string, T> GetParser(string type)
         {
             switch (type)
             {
                 case nameof(Boolean):
-                    return v => bool.Parse(v);
+                    return v => Convert(bool.Parse(v));
+                case nameof(Int32):
+                    return v => (T)(object)int.Parse(v);
                 default:
                     throw new ArgumentException(nameof(type));
             }
         }
 
-        public bool Equals(CodeStyleOption<T> other)
-        {
-            return EqualityComparer<T>.Default.Equals(Value, other.Value) &&
-                   Notification == other.Notification;
-        }
-
-        public override bool Equals(object obj)
+        private static T Convert(bool b)
         {
-            var codeStyle = obj as CodeStyleOption<T>;
-            if (codeStyle == null)
+            if (typeof(T) == typeof(bool))
             {
-                return false;
+                return (T)(object)b;
             }
 
-            return ((IEquatable<CodeStyleOption<T>>)this).Equals(codeStyle);
+            var enumValues = (T[])Enum.GetValues(typeof(T));
+            return b ? enumValues[0] : enumValues[1];
```

Comment:
I've also made it so that we just go and find the item with the 0 and 1 value instead of assumign they're the 0th and 1st in that array.